File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0;
        int r = 1;
        int max = 0;

        while (r < prices.size() && r >= l) {
            max = max > prices[r] - prices[l] ? max : prices[r] - prices[l];
            if (prices[r] > prices[l]) {
                r += 1;
            }
            else {
                l = r;
                r += 1;
            }
        }
        return max;
    }
};

File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        m = float("-inf")

        def dfs(root):
            nonlocal m
            if root:
                max_left = dfs(root.left)
                max_right = dfs(root.right)
                temp = max(root.val, root.val+max_left, root.val+max_left+max_right, root.val+max_right)
                max_val = max(root.val, root.val+max_left, root.val+max_right)
                if temp > m:
                    m = temp
                return max_val
            return 0
        
        dfs(root)
        return m